src/EventBus/Middleware/CallbackOnFirstPublicationMiddleware.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\EventBus\Middleware;
+
+use Broadway\Domain\DomainEventStream;
+use Closure;
+use CultuurNet\UDB3\EventBus\EventBusMiddleware;
+
+final class CallbackOnFirstPublicationMiddleware implements EventBusMiddleware
+{
+    private int $publicationCount = 0;
```

Comment:
I wonder if a "state" boolean would make more sense here?

Reply:
I don't really have an opinion on this, I guess a boolean might be safer if we do a super large amount of publications (maybe in a long-running CLI command). Can always be refactored in a follow-up PR :)